### ISSUE TITLE
Support for IPv6

### ISSUE THREAD

**Issue by Fuchs-David** [SWI-Prolog/packages-clib#24]
Current socket implementation lacks IPv6 support.

This will become a major concern for anyone dealing with the US Government, because they will be obligated (within 2 years from now) to refuse any devices or applications that fail in their IPv6-only testing environment. 

USGv6 testing program:
https://www.iol.unh.edu/testing/ip/ipv6/usgv6
Article describing the change to USGv6:
https://www.electronicdesign.com/industrial-automation/what-s-new-latest-usgv6-profile-update

**Comment by JanWielemaker** [comment]
I fully agree.  This thing waits for someone being sufficiently motivated to contribute the code or make it happen some other way.  For now, I don't have immediate use for it and there are plenty of, for me and direct stake holders, more urgent things to solve.

**Comment by ghost** [comment]
Simple connectivity testing can be done via:

You as a client:
https://ipv6-test.com/

The server you are accessing:
https://ipv6-test.com/validate.php
